beacon-chain/blockchain/forkchoice/service.go

```diff
@@ -41,18 +41,20 @@ type Store struct {
 	checkpointStateLock  sync.Mutex
 	seenAtts             map[[32]byte]bool
 	seenAttsLock         sync.Mutex
+	shouldArchive        bool
 }
 
 // NewForkChoiceService instantiates a new service instance that will
 // be registered into a running beacon node.
-func NewForkChoiceService(ctx context.Context, db db.Database) *Store {
+func NewForkChoiceService(ctx context.Context, db db.Database, shouldArchive bool) *Store {
```

Comment:
Instead of passing flag as an argument, can we use a global flag? (similar to what we did with feature flag)
With the earlier discussions, it concluded that passing flags as arguments in generally not preferred

Reply:
The issue is that the archiver isnt under a global flag. Other than our feature flags, all our other flags use the cli context including the archiver.